### PR TITLE
Pin Pagy v5 & update govuk-components to v3.3

### PR DESCRIPTION
**Pull request by lozette**
## Checklist

- [x] Attach this pull request to the appropriate card in DevOps.
- [ ] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -154,10 +154,12 @@ GEM
     ffi (1.15.5)
     globalid (1.1.0)
       activesupport (>= 5.0)
-    govuk-components (3.3.0)
+    govuk-components (3.1.1)
```
Comment (mec) [inline Gemfile.lock:157]: This is going backwards? Is that right?

**Review by mec — CHANGES_REQUESTED** [review]
Is there a reason `govuk-components` goes back to 3.1.1 here?

File: Gemfile.lock (hunk 1 above)
Comment (lozette) [reply Gemfile.lock:157]: Ooh I didn't even notice! No that's not right

Comment (lozette) [reply Gemfile.lock:157]: Oh no 
```
Bundler could not find compatible versions for gem "pagy":
  In Gemfile:
    pagy (~> 6.1)

    govuk-components (~> 3.3) was resolved to 3.3.0, which depends on
      pagy (~> 5.10.1)
```
Looks like we have to hold off updating pagy until govuk-components catches up?

Comment (mec) [reply Gemfile.lock:157]: Aha! That is what I was worried about when pagy came in via Renovate!

Okay, shall pin Pagy to the compatible version, we can say why in the commit?

**Pull request closed without merging**